server/pkg/sqlutil/tenant.go

```diff
@@ -0,0 +1,58 @@
+package sqlutil
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/kubeshop/tracetest/server/http/middleware"
+)
+
+func Tenant(ctx context.Context, query string, params ...any) (string, []any) {
+	tenantID := TenantID(ctx)
+	if tenantID == nil {
+		return query, params
+	}
+
+	prefix := getQueryPrefix(query)
+
+	paramNumber := len(params) + 1
+	condition := fmt.Sprintf(" %s tenant_id = $%d", prefix, paramNumber)
+
+	return query + condition, append(params, tenantID)
+}
+
+func TenantWithPrefix(ctx context.Context, query string, prefix string, params ...any) (string, []any) {
+	tenantID := TenantID(ctx)
+	if tenantID == nil {
+		return query, params
+	}
+
+	queryPrefix := getQueryPrefix(query)
+	paramNumber := len(params) + 1
+	condition := fmt.Sprintf(" %s %stenant_id = $%d)", queryPrefix, prefix, paramNumber)
+
+	return query + condition, append(params, tenantID)
+}
+
+func TenantID(ctx context.Context) *string {
+	tenantID := ctx.Value(middleware.TenantIDKey)
+
+	if tenantID == "" || tenantID == nil {
```

Comment:
You are creating each resource with:

```sql
COLUMN tenant_id uuid DEFAULT '00000000-0000-0000-0000-000000000000';
``` 

Shouldn't you inject `00000000-0000-0000-0000-00000000000` into the context when user passes no tenant information?

Reply:
I removed the default value because that might cause problems with backward compatibility, right now I think is better to have it always be `NULL`. Good catch!